src/sqlfluff/api/simple.py

```diff
@@ -123,8 +123,17 @@ def fix(
     linter = Linter(config=cfg)
 
     result = linter.lint_string_wrapped(sql, fix=True)
-    fixed_string = result.paths[0].files[0].fix_string()[0]
-    return fixed_string
+    fix_even_unparsable = cfg.get("fix_even_unparsable")
+    should_fix = fix_even_unparsable
+    if not should_fix:
```

Comment:
I'm very confused by this.

1) Why do we need two variables (`fix_even_unparsable` and `should_fix`)?
2) `if not should_fix` is doing my head in. Any way to rephrase without the `not`? Especially as it seems to be a variable created just for this.

Reply:
It's because there are not one but *two* ways that `should_fix` can be set to `True`, either initially, based on `fix_even_unparsable`, or inside the `if` statement. Basically, we "should fix" the file if either:
* `fix_even_unparsable` is set **OR**
* The file has no parse errors. The code inside the `if` checks for this.

There are other ways to structure this code that may be clearer, but I think they would involve duplicating some of the code.

Reply:
Well the way it’s currently written is a bit of an optimisation - it only calls the unparsable check if it needs to. So those using the force option (that we want to discourage) save that check.

Given that we want to discourage that force nature, I’m not really inclined to optimise for them, and would favour code readability over that.

So I would say do the check and then after, if no parse errors OR flag is set, do the fix.

WDYT?

Reply:
I agree the current code looks odd when you initially see it, but I'm also trying to consider if you're a maintainer and you take a closer look, you're likely to ask, "Why is it doing all this useless work to count errors when it doesn't need to? Maybe I'm missing something. I'm afraid to touch this code."

I'm making a slight change that you may like better.